File: app/src/main/java/com/zzzmode/appopsx/ui/BaseActivity.java
package com.zzzmode.appopsx.ui;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import com.zzzmode.appopsx.ui.analytics.ATracker;
import com.zzzmode.appopsx.ui.core.LangHelper;
import com.zzzmode.appopsx.ui.core.SpHelper;

/**
 * Created by zl on 2017/1/7.
 */

public class BaseActivity extends AppCompatActivity {

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    LangHelper.updateLanguage(this);
    super.onCreate(savedInstanceState);
    AppCompatDelegate.setDefaultNightMode(SpHelper.getThemeMode(this));

  }

  @Override
  protected void attachBaseContext(Context newBase) {
    super.attachBaseContext(LangHelper.attachBaseContext(newBase));
  }

  @Override
  protected void onResume() {
    super.onResume();
    ATracker.onResume(this);
  }

  @Override
  protected void onPause() {
    super.onPause();
    ATracker.onPause(this);
  }
}

File: app/src/main/kotlin/com/zzzmode/appopsx/ui/BaseActivity.kt
package com.zzzmode.appopsx.ui

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.app.AppCompatDelegate
import com.zzzmode.appopsx.ui.analytics.ATracker
import com.zzzmode.appopsx.ui.core.LangHelper
import com.zzzmode.appopsx.ui.core.SpHelper

/**
 * Created by zl on 2017/1/7.
 */

open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        LangHelper.updateLanguage(this)
        super.onCreate(savedInstanceState)
        AppCompatDelegate.setDefaultNightMode(SpHelper.getThemeMode(this))

    }

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(LangHelper.attachBaseContext(newBase))
    }

    override fun onResume() {
        super.onResume()
        ATracker.onResume(this)
    }

    override fun onPause() {
        super.onPause()
        ATracker.onPause(this)
    }
}
